From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: IPECTER <ipectert@gmail.com>
Date: Fri, 19 Sep 2025 18:00:27 +0900
Subject: [PATCH] SIMD support

Project: Pufferfish (https://github.com/pufferfish-gg/Pufferfish)
Author: Kevin Raneri <kevin.raneri@gmail.com>
License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)

Project: Gale (https://github.com/GaleMC/Gale)
Author: Martijn Muijsers <martijnmuijsers@live.nl>
License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)

Description:
* This patch is based on Gale's "SIMD support" patch.

diff --git a/net/minecraft/server/dedicated/DedicatedServer.java b/net/minecraft/server/dedicated/DedicatedServer.java
index 36ef01826f4d4736e686d70389577305710d710b..fd173f547887c9ea74c8f59af8254ddd3112b1aa 100644
--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -200,6 +200,12 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         // Purpur end - Purpur config files
         com.destroystokyo.paper.VersionHistoryManager.INSTANCE.getClass(); // Paper - load version history now
 
+        // GPlazma start - SIMD support
+        try {
+            org.plazmamc.plazma.simd.SIMDDetection.initialize();
+        } catch (Throwable ignored) {}
+        // Plazma end - SIMD support
+
         this.setPvpAllowed(properties.pvp);
         this.setFlightAllowed(properties.allowFlight);
         this.setMotd(properties.motd);
